sphinx/source/docs/user_guide/concepts.rst

```diff
@@ -124,145 +123,157 @@ functions. Scripts that output with these typically look something like:
 
     show(p)
 
-If this script is called ``foo.py`` then executing ``python foo.py`` will
-result in an HTML file ``output.html`` being generated with the line plot.
-These functions are often useful in interactive settings, or for creating
-standalone Bokeh documents to serve from (Flask, Django, etc.) web
-applications.
+This script generates an HTML file called ``output.html`` that contains a line
+plot. You can execute it with ``python foo.py``, where ``foo.py`` is the name
+of the script.
 
-However, Bokeh also comes with a powerful command line tool ``bokeh`` that
-can also be used to generate various kinds of output:
+These functions are often useful in interactive settings or for creating
+standalone Bokeh documents to serve from backend web applications.
 
-``bokeh html``
-    Create standalone HTML documents from any kind of Bokeh application
-    source: e.g., python scripts, app directories, JSON files, and others.
+Another way to generate various kinds of output is by using Bokeh's powerful
+:ref:`command line tool <userguide_cli>`.
 
-``bokeh json``
-    Generate a serialized JSON representation of a Bokeh document from any
-    kind of Bokeh application source.
+.. _userguide_interfaces:
 
-``bokeh serve``
-    Publish Bokeh documents as interactive web applications.
+Interfaces
+----------
 
-An advantage of using the ``bokeh`` command is that the code you write does not
-have to specify any particular output method or format. You can write *just the
-visualization code* once, and decide later to output in different ways. The
-above example would be simplified to:
+Bokeh provides a simple and intuitive interface for users like data scientists
+and domain experts who do not wish to be distracted by complex details of the
+software. At the same time, Bokeh also caters to people such as application
+developers and software engineers who may want more control or access to more
+sophisticated features.
 
-.. code-block:: python
+Because of this, Bokeh takes a layered approach and offers different programming
+interfaces appropriate to different users.
 
-    from bokeh.plotting import figure, curdoc
+This section provides an overview of the two interfaces that you can use: the
+mid-level :ref:`bokeh.plotting interface <userguide_interfaces_plotting>` and
```

Comment:
I wonder if it is time to find a better description. "mid-level" was applied back when `bokeh.charts` was intended to be "high-level" and part of the core Bokeh library. But `bokeh.charts` was abandoned. Perhaps we could just say `bokleh.plotting` is the "primary" interface? I'm still content to refer to `bokeh.models` and the "low-level" because that carries the appropriate connotations.

Reply:
I was wondering about that and meant to ask - thanks for adding some background! "primary" sounds perfect to me!